### PR TITLE
Multiple clicks in massaction dropdown generates multiple post request

### LINKED ISSUE #37997: [Issue] Multiple clicks in massaction dropdown generates multiple post request
This issue is automatically created based on existing pull request: magento/magento2#37996: Multiple clicks in massaction dropdown generates multiple post request

---------

### Preconditions (*)
Tested on this versions:
    1. Magento 2.4.5-p1
    2. Magento 2.4.6-p1
    3. Magento 2.4.6-p2

### Description (*)

When we go to the Sales section, specifically the Orders page in Magento 2, and select an order, you have the option to perform actions in bulk, which we call "mass actions." For example, let's say I click the "Hold" action twice. This action triggers two POST requests, which are recorded in the Apache access log (screenshots 1 and 2). Now, if you were to click any action three times, it would result in the generation of three simultaneous POST requests. This behavior isn't limited to the Sales Orders page. It applies to other areas in Magento 2 as well. For instance, if you're working with the product grid and perform a mass action there, like clicking "Update Attributes" three times, it generates three POST requests. You can see these requests in the screenshots I’ve shared (3 and 4). In simpler terms, when you use mass actions in Magento 2, like holding orders or updating product attributes, multiple requests can be sent to the server simultaneously if you click the action multiple times quickly. This behavior can be observed not only in the Sales Orders section but also in other parts of the system, leading to multiple requests being processed at the same time.

### Screenshot-1
I clicked the "Hold" action twice in the order grid, resulting in the generation of two POST requests.
image

### Screenshot-2
image

### Screenshot-3
Similarly, I clicked the "Update Attributes" action three times, leading to the generation of three POST requests.
image

### Screenshot-4
image

### Consequences
In our scenario, we have a module that handles the creation of shipments through a mass action. While there is a safeguard in place to check for existing shipments during the creation process, the issue arises when users repeatedly click the action, triggering multiple POST requests. This can inadvertently result in the generation of multiple shipments, the actual quantity of items shipped may be lower than the salable quantity.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by Varshal24**
### Preconditions (*)
Tested on this versions:
    1. Magento 2.4.5-p1
    2. Magento 2.4.6-p1
    3. Magento 2.4.6-p2

### Description (*)

When we go to the Sales section, specifically the Orders page in Magento 2, and select an order, you have the option to perform actions in bulk, which we call "mass actions." For example, let's say I click the "Hold" action twice. This action triggers two POST requests, which are recorded in the Apache access log (screenshots 1 and 2). Now, if you were to click any action three times, it would result in the generation of three simultaneous POST requests. This behavior isn't limited to the Sales Orders page. It applies to other areas in Magento 2 as well. For instance, if you're working with the product grid and perform a mass action there, like clicking "Update Attributes" three times, it generates three POST requests. You can see these requests in the screenshots I’ve shared (3 and 4). In simpler terms, when you use mass actions in Magento 2, like holding orders or updating product attributes, multiple requests can be sent to the server simultaneously if you click the action multiple times quickly. This behavior can be observed not only in the Sales Orders section but also in other parts of the system, leading to multiple requests being processed at the same time.

### Screenshot-1
I clicked the "Hold" action twice in the order grid, resulting in the generation of two POST requests.
image

### Screenshot-2
image

### Screenshot-3
Similarly, I clicked the "Update Attributes" action three times, leading to the generation of three POST requests.
image

### Screenshot-4
image

### Consequences
In our scenario, we have a module that handles the creation of shipments through a mass action. While there is a safeguard in place to check for existing shipments during the creation process, the issue arises when users repeatedly click the action, triggering multiple POST requests. This can inadvertently result in the generation of multiple shipments, the actual quantity of items shipped may be lower than the salable quantity.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#37997: Multiple clicks in massaction dropdown generates multiple post request

### CODE REVIEW

**Comment by Varshal24** [comment]
@magento run all tests

**Comment by engcom-Bravo** [comment]
@magento create issue

**Comment by andrewbess** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Static Tests, WebAPI Tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Static Tests, WebAPI Tests

**Comment by engcom-Hotel** [comment]
@magento run Static Tests

**Comment by engcom-Hotel** [comment]
@magento run Static Tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B, Integration Tests, WebAPI Tests

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @Varshal24,

Thanks for the contribution!

Changes seem good to me, but I agree with the @rogyar here https://github.com/magento/magento2/pull/37996#issuecomment-1913639818, we should close the dropdown whenever we click on any option.

So please do the needful and also please solve the `Static Test` failures.

Thanks

**Comment by Varshal24** [comment]
@magento run all tests

**Review by engcom-Hotel — APPROVED** [review]
Hello @Varshal24,

Thanks for making the changes!

Changes looks good to me, approving the PR.

Failed tests seem flaky to me.

Thanks

**Comment by engcom-Delta** [comment]
### :heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** 

**Manual testing scenario:**



**Before:** :heavy_multiplication_x: 

Screenshot 2024-02-14 at 8 52 16 PM

Screenshot 2024-02-14 at 8 53 07 PM

Screenshot 2024-02-14 at 10 07 39 PM

Screenshot 2024-02-14 at 10 10 39 PM

Screenshot 2024-02-14 at 10 11 03 PM

Screenshot 2024-02-14 at 10 17 02 PM

Screenshot 2024-02-14 at 10 17 25 PM

Screenshot 2024-02-14 at 10 17 42 PM

Screenshot 2024-02-14 at 10 18 07 PM

Screenshot 2024-02-14 at 10 18 26 PM

Screenshot 2024-02-14 at 10 22 30 PM


**After:** :heavy_check_mark:  

Screenshot 2024-02-14 at 10 45 23 PM

Screenshot 2024-02-14 at 10 45 47 PM

Screenshot 2024-02-14 at 10 46 47 PM

Screenshot 2024-02-14 at 10 47 48 PM

Screenshot 2024-02-14 at 10 48 04 PM

Screenshot 2024-02-14 at 10 50 34 PM


Builds are failed. Hence, moving this PR to **Extended Testing.**

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, Integration Tests, WebAPI Tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests EE

**Comment by engcom-Charlie** [comment]
Functional EE failures are different in latest 2 builds with the same code base. It seems to be flaky.

**Run 1:** https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/37996/61d1d970a477785897be194f134f2beb/Functional/allure-report-ee/index.html#categories
image


**Run2:** https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/37996/4938a0786cbc61df0f0c34ad3e546f35/Functional/allure-report-ee/index.html
image


Functional B2B failures are not part of this PR and seems to be flaky.
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/37996/3839e641329ef71ea99398b848e6fc4c/Functional/allure-report-b2b/index.html#categories

**Comment by engcom-Charlie** [comment]
@magento run Static Tests

**Comment by engcom-Charlie** [comment]
@magento run Semantic Version Checker
